SIGSEGV panic in 1.18 Agent DNSZombieMappings

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

equal or higher than v1.18.2 and lower than v1.19.0

### What happened?

When upgrading our clusters from 1.17 to 1.18 multiple nodes had the Agent crashing on startup with the following error:
```
{"time":"2025-10-15T22:01:51.347740156Z","level":"info","msg":"Restoring checkpointed local identities","module":"agent.controlplane.identity.identity","file-path":"/var/run/cilium/state/local_allocator_state.json","count":15}
{"time":"2025-10-15T22:01:51.347871996Z","level":"info","msg":"Processing identity update","module":"agent.controlplane.policy","policyID.Added":[16777593,16777652,16777222,16777224,16777218,16777223,16777663,16777227,16777228,16777665,16777233,16777659,16777221,16777230,16777231],"policyID.Deleted":null}
{"time":"2025-10-15T22:01:51.348159482Z","level":"info","msg":"Removed map pin, recreating and re-pinning map","bpfMapPath":"cilium_ipcache_v2","bpfMapName":"cilium_ipcache_v2"}
{"time":"2025-10-15T22:01:51.348935006Z","level":"info","msg":"Reading old endpoints...","module":"agent.controlplane.daemon"}
{"time":"2025-10-15T22:01:51.350165182Z","level":"info","msg":"Incremental policy update: waiting for endpoint notifications to complete","module":"agent.controlplane.policy","count":1}
{"time":"2025-10-15T22:01:51.350217697Z","level":"info","msg":"Incremental policy update: triggering UpdatePolicyMaps for all endpoints","module":"agent.controlplane.policy"}
{"time":"2025-10-15T22:01:51.372703852Z","level":"info","msg":"restored DNS history from endpoint","module":"agent.controlplane.fqdn.namemanager","endpointID":3519,"count":13,"numAddresses":478}
{"time":"2025-10-15T22:01:51.375396725Z","level":"info","msg":"restored DNS history from endpoint","module":"agent.controlplane.fqdn.namemanager","endpointID":3162,"count":6,"numAddresses":242}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x55c3d0]

goroutine 1 [running]:
log/slog.(*Logger).Handler(...)
	/usr/local/go/src/log/slog/logger.go:121
log/slog.(*Logger).Enabled(0x0?, {0x4e3cd70?, 0x815a8e0?}, 0xfa612c47c688?)
	/usr/local/go/src/log/slog/logger.go:168 +0x20
log/slog.(*Logger).log(0x0, {0x4e3cd70, 0x815a8e0}, 0x4, {0x40024b6680, 0xca}, {0x0, 0x0, 0x0})
	/usr/local/go/src/log/slog/logger.go:241 +0x48
log/slog.(*Logger).Warn(...)
	/usr/local/go/src/log/slog/logger.go:219
github.com/cilium/cilium/pkg/fqdn.(*DNSZombieMappings).GC(0x4003e789c0)
	/go/src/github.com/cilium/cilium/pkg/fqdn/cache.go:1036 +0x644
github.com/cilium/cilium/pkg/fqdn/namemanager.(*manager).RestoreCache(0x4001248600, 0x400131c930)
	/go/src/github.com/cilium/cilium/pkg/fqdn/namemanager/gc.go:218 +0x33c
github.com/cilium/cilium/daemon/cmd.newDaemon({0x4e3cfd0, 0x4002506500}, 0x4001060a70, 0x40008cc008)
	/go/src/github.com/cilium/cilium/daemon/cmd/daemon.go:463 +0xd88
github.com/cilium/cilium/daemon/cmd.newDaemonPromise.func1({0x4002e99748?, 0xa059d8?})
	/go/src/github.com/cilium/cilium/daemon/cmd/daemon_main.go:1438 +0xa0
github.com/cilium/hive/cell.Hook.Start(...)
	/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/cell/lifecycle.go:43
github.com/cilium/hive/cell.(*DefaultLifecycle).Start(0x40002d9770, 0x4000522cf0, {0x4e3d040?, 0x400053c000?})
	/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/cell/lifecycle.go:128 +0x2a0
github.com/cilium/hive.(*Hive).Start(0x4000d10e10, 0x4000522cf0, {0x4e3d040, 0x400053c000})
	/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:359 +0xe0
github.com/cilium/hive.(*Hive).Run(0x4000d10e10, 0x4000522cf0)
	/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:231 +0x70
github.com/cilium/cilium/daemon/cmd.NewAgentCmd.func1(0x4000a62c08, {0x47156ca?, 0x4?, 0x471556a?})
	/go/src/github.com/cilium/cilium/daemon/cmd/root.go:52 +0x1d4
github.com/spf13/cobra.(*Command).execute(0x4000a62c08, {0x40000b4190, 0x17, 0x17})
	/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:1019 +0x82c
github.com/spf13/cobra.(*Command).ExecuteC(0x4000a62c08)
	/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:1148 +0x384
github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:1071
github.com/cilium/cilium/daemon/cmd.Execute(0x4978060?)
	/go/src/github.com/cilium/cilium/daemon/cmd/root.go:90 +0x1c
main.main()
	/go/src/github.com/cilium/cilium/daemon/main.go:15 +0x28
```

I believe this is caused by `ep.getLogger()` returning `nil` here when the endpoint is being restored:
https://github.com/cilium/cilium/blob/586a7c5fbded51f0c96c0dbe7239934a4ecae6a8/pkg/endpoint/restore.go#L563
So `DNSZombieMappings` ends up with a `nil` logger. 

The problem can be clearly seen here, first `ep.UnmarshalJSON()` is called: 
https://github.com/cilium/cilium/blob/586a7c5fbded51f0c96c0dbe7239934a4ecae6a8/pkg/endpoint/endpoint.go#L955 
and only afterwards the logger is actually initialized:
https://github.com/cilium/cilium/blob/586a7c5fbded51f0c96c0dbe7239934a4ecae6a8/pkg/endpoint/endpoint.go#L997

### How can we reproduce the issue?

-

### Cilium Version

```
Daemon: 1.18.2 e359538840 2025-09-25T14:38:13+02:00 go version go1.24.7 X:boringcrypto linux/arm64
```

### Kernel Version

-

### Kubernetes Version

-

### Regression

_No response_

### Sysdump

_No response_

### Relevant log output

```shell

```

### Anything else?

_No response_

### Cilium Users Document

- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/pkg/endpoint/restore.go
+++ b/pkg/endpoint/restore.go
@@ -560,7 +560,7 @@ func (ep *Endpoint) UnmarshalJSON(raw []byte) error {
 		Labels:     labels.NewOpLabels(),
 		Options:    option.NewIntOptions(&EndpointMutableOptionLibrary),
 		DNSHistory: fqdn.NewDNSCacheWithLimit(option.Config.ToFQDNsMinTTL, option.Config.ToFQDNsMaxIPsPerHost),
-		DNSZombies: fqdn.NewDNSZombieMappings(ep.getLogger(), option.Config.ToFQDNsMaxDeferredConnectionDeletes, option.Config.ToFQDNsMaxIPsPerHost),
+		DNSZombies: fqdn.NewDNSZombieMappings(ep.Logger(subsystem), option.Config.ToFQDNsMaxDeferredConnectionDeletes, option.Config.ToFQDNsMaxIPsPerHost),
 	}
 	if err := json.Unmarshal(raw, restoredEp); err != nil {
 		return fmt.Errorf("error unmarshaling serializableEndpoint from base64 representation: %w", err)

